README.md

```diff
@@ -1,24 +1,101 @@
- q.setDocumentLang("en");                        // only search for posts in english
- Result result = q.makeRequest("spotify");       // find posts containing the word spotify
- for (Post p : result.getPosts())
-     System.out.println(p.getUrl());             // print the url for each post
+// create Query object using your API key
+Query query = new Query(apiKey);
+// find posts containing word "spotify" in English language created from startDate to endDate
+Result result = query.makeRequest("spotify", Language.English, startDate, endDate);
+// Get the total number of matches returned
+int numberOfMatchesReturned = result.getNumberOfMatchesReturned();
+// Get the total number of matches to the search query
+int numberOfMatchesTotal = result.getNumberOfMatchesTotal();
+// Get number of seconds it took to execute query
+double secondsElapsed = result.getSecondsElapsed();
+for (Post p : result.getPosts())
+    System.out.println(p.getUrl());             // print url for each post
+    System.out.println(p.getTitle());           // print title for each post
 ```
 
-Later you can repeat the search for any new posts that have come in since your last request:
+Later you can repeat the search for any new posts with same or other parameters:
 
 ```Java
- q.setStartTime(result.getNewestPost());         // only search for posts newer than the last result
- result = q.makeRequest("spotify");              // request new posts containing the word spotify
+// find posts containing word "spotify" in English language created from startDate (up to now)
+query.makeRequest("spotify", Language.English, startDate);
+// find posts containing word "spotify" in English language (all posts, without any time limits)
+query.makeRequest("spotify", Language.English);
+// find posts containing word "spotify" (all posts, withou time and language limits)
+query.makeRequest("spotify");
 ```
 
 A Query object can be reused for several requests
+
+## Requirements
+
+* API key, contact [sales](mailto:sales@twingly.com) via [twingly.com](https://www.twingly.com/try-for-free/) to get one
+* Java:
+    * v 1.7
+    * v 1.8
+
+
+## Development
+
+### Tests
+
+Make sure you have all the dependencies
+```gradle
+gradlew assemble
+```
+Run the tests
+```
+gradlew check
+```
+
+### Exception handling
+
+For now, any exception is wrapped in TwinglyException class.
+
+### JavaDoc generation
+
+To generate java docs just run gradle task `javadoc`:
+```
+gradle javadoc
+```
+This will save javadocs in `docs` folder under build-folder.
+
+
+### One-jar generation
+
+To generate one fat jar library with all compile dependencies just run gradle task `generateOneJar`:
+```
+gradle generateOneJar
+```
+This will save JAR file with all needed dependencies in `build/libs/` with `-all` suffix. (like `twingly-search-api-java-all-0.0.1.jar`)
+
+## License
+
+The MIT License (MIT)
+
+Copyright (c) 2016 Twingly AB
```

Comment:
There's a file `LICENSE.txt` that can be removed. The Copyright in this file could be changed to 2012-2016.